pkg/agent/util/sriov_utils.go

```diff
@@ -0,0 +1,140 @@
+package util
+
+import (
+        "fmt"
+        "context"
+        "net"
+        "os"
+        "time"
+        "github.com/containernetworking/cni/pkg/skel"
+        "github.com/containernetworking/cni/pkg/types"
+        "k8s.io/kubernetes/pkg/kubelet/util"
+        "k8s.io/kubernetes/pkg/kubelet/apis/podresources"
+        podresourcesv1alpha1 "k8s.io/kubernetes/pkg/kubelet/apis/podresources/v1alpha1"
+        sriov_cni_utils "github.com/k8snetworkplumbingwg/sriov-cni/pkg/utils"
+        "k8s.io/klog"
+)
+
+const (
+        // Kubelet internal cgroup name for node allocatable cgroup.
+        defaultNodeAllocatableCgroup = "kubepods"
+        // defaultPodResourcesPath is the path to the local endpoint serving the podresources GRPC service.
+        defaultPodResourcesPath    = "/var/lib/kubelet/pod-resources"
+        defaultPodResourcesTimeout = 10 * time.Second
+        defaultPodResourcesMaxSize = 1024 * 1024 * 16 // 16 Mb
+)
+
+type KubeletPodResources struct {
+        resources []*podresourcesv1alpha1.PodResources
+}
+
+// K8sArgs is the valid CNI_ARGS used for Kubernetes
+type K8sArgs struct {
+        types.CommonArgs
+        IP                         net.IP
+        K8S_POD_NAME               types.UnmarshallableString
+        K8S_POD_NAMESPACE          types.UnmarshallableString
+        K8S_POD_INFRA_CONTAINER_ID types.UnmarshallableString
+}
+
+// GetK8sArgs gets k8s related args from CNI args
+func GetK8sArgs(args *skel.CmdArgs) (*K8sArgs, error) {
+        k8sArgs := &K8sArgs{}
+
+        err := types.LoadArgs(args.Args, k8sArgs)
+        if err != nil {
+                return nil, err
+        }
+
+        return k8sArgs, nil
+}
+
+func GetNodeDevices() (*podresourcesv1alpha1.ListPodResourcesResponse, error) {
```

Comment:
I feel these K8s calls should not be in agent/util/. Maybe we should just move them to a separate file in agent/cniserver/ like sriov.go, or secondary_interface.go if we choose to put all secondary interface code (including: configureMultipleInterfaces) there.

Reply:
Done